hal/audio_extn/cirrus_sony.c

```diff
@@ -611,6 +611,13 @@ static int cirrus_play_silence(int seconds) {
     if (!uc_info_rx) {
         return -ENOMEM;
     }
+
+    while (!adev->primary_output) {
+        ALOGE("Still no primary_output!");
+        // TODO: Perhaps wait on a condvar like spkr_prot?
+        usleep(1000);
```

Comment:
Are you sure that this really works? Is this stuff being executed in a pthread while the HAL continues the other part of the init?!??!?

Because in my head again, the solution for this was to spawn a silence-calibration pthread (but that was WAAAAAAAAAY MORE complicated than what you just did, you know, sync everything, mutexes blah, make sure that nothing plays before this function succeeds blahblah, big mess)

Reply:
Why don't we take a look in `spkr_protection`?

https://github.com/sonyxperiadev/vendor-qcom-opensource-audio-hal-primary-hal/blob/13b245e5edfadcb235e2ac32c61b0eee6328c74d/hal/audio_extn/spkr_protection.c#L1275-L1278

https://github.com/sonyxperiadev/vendor-qcom-opensource-audio-hal-primary-hal/blob/13b245e5edfadcb235e2ac32c61b0eee6328c74d/hal/audio_extn/spkr_protection.c#L1123-L1131

Hmm, so we have a condvar...

That is only ever signaled here:

https://github.com/sonyxperiadev/vendor-qcom-opensource-audio-hal-primary-hal/blob/13b245e5edfadcb235e2ac32c61b0eee6328c74d/hal/audio_extn/spkr_protection.c#L1930-L1935

That's only ever called from deinit... and...

https://github.com/sonyxperiadev/vendor-qcom-opensource-audio-hal-primary-hal/blob/13b245e5edfadcb235e2ac32c61b0eee6328c74d/hal/audio_extn/spkr_protection.c#L2014

Which is called from

https://github.com/sonyxperiadev/vendor-qcom-opensource-audio-hal-primary-hal/blob/13b245e5edfadcb235e2ac32c61b0eee6328c74d/hal/audio_extn/audio_extn.c#L6570

Reply:
By the way, notice that `cirrus_do_calibration` already runs asynchronously in a thread.